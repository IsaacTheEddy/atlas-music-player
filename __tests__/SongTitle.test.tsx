import { render } from "@testing-library/react";
import { test, expect } from "vitest";
import SongTitle from "../src/components/SongTitle";
import {
  currentSongContext,
  useCurrentSongContext,
} from "../src/components/context";

test("Non empty Song Title, and Artist name passed in", () => {
  let { container } = render(
    <currentSongContext.Provider
      value={{
        playlist: [],
        currentSong: {
          id: "a6pkp78whsyqdvpb5dxn64ss",
          title: "Fading Shadows",
          artist: "The Emberlight",
          genre: "Alternative Rock",
          duration: 194,
          cover:
            "https://utfs.io/f/E9fJnaKtTy1bV09oPkISuh6fWpNsTRlAk1Qj9yqnVzCi32BL",
          song: "https://utfs.io/f/E9fJnaKtTy1ba1N97yFrtlodEaJI0m4wGY9KgyPiUvCf8hMp",
          isPlaying: false,
        },
      }}
    >
      <SongTitle />
    </currentSongContext.Provider>,
  );

  expect(container).toMatchSnapshot();
});
test("empty Song Title, Non Empty Artist name passed in", () => {
  let { container } = render(
    <currentSongContext.Provider
      value={{
        playlist: [],
        currentSong: {
          id: "a6pkp78whsyqdvpb5dxn64ss",
          artist: "The Emberlight",
          genre: "Alternative Rock",
          duration: 194,
          cover:
            "https://utfs.io/f/E9fJnaKtTy1bV09oPkISuh6fWpNsTRlAk1Qj9yqnVzCi32BL",
          song: "https://utfs.io/f/E9fJnaKtTy1ba1N97yFrtlodEaJI0m4wGY9KgyPiUvCf8hMp",
          isPlaying: false,
        },
      }}
    >
      <SongTitle />
    </currentSongContext.Provider>,
  );

  expect(container).toMatchSnapshot();
});
test("empty Song Title, Empty Artist name passed in", () => {
  let { container } = render(
    <currentSongContext.Provider
      value={{
        playlist: [],
        currentSong: {
          id: "a6pkp78whsyqdvpb5dxn64ss",
          genre: "Alternative Rock",
          duration: 194,
          cover:
            "https://utfs.io/f/E9fJnaKtTy1bV09oPkISuh6fWpNsTRlAk1Qj9yqnVzCi32BL",
          song: "https://utfs.io/f/E9fJnaKtTy1ba1N97yFrtlodEaJI0m4wGY9KgyPiUvCf8hMp",
          isPlaying: false,
        },
      }}
    >
      <SongTitle />
    </currentSongContext.Provider>,
  );

  expect(container).toMatchSnapshot();
});
