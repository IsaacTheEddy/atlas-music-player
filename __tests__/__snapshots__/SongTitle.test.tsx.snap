// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Non empty Song Title, and Artist name passed in 1`] = `
<div>
  <div
    class="flex-nowrap pt-6"
  >
    <h2
      class="font-sans text-2xl font-bold"
    >
      Fading Shadows
    </h2>
    <p
      class="font-sans font-bold text-Mocha/75 dark:text-Lightsoftie/75"
    >
      The Emberlight
    </p>
  </div>
</div>
`;

exports[`empty Song Title, Empty Artist name passed in 1`] = `
<div>
  <div
    class="flex-nowrap pt-6"
  >
    <h2
      class="font-sans text-2xl font-bold"
    >
      Song Title
    </h2>
    <p
      class="font-sans font-bold text-Mocha/75 dark:text-Lightsoftie/75"
    >
      Artist
    </p>
  </div>
</div>
`;

exports[`empty Song Title, Non Empty Artist name passed in 1`] = `
<div>
  <div
    class="flex-nowrap pt-6"
  >
    <h2
      class="font-sans text-2xl font-bold"
    >
      Song Title
    </h2>
    <p
      class="font-sans font-bold text-Mocha/75 dark:text-Lightsoftie/75"
    >
      The Emberlight
    </p>
  </div>
</div>
`;
